### ISSUE TITLE
Messages on the message tab (`/messages`) should be filtered by language

### ISSUE THREAD

**Issue by andreaskoepf** [LAION-AI/Open-Assistant#962]
Currently the messages show in on the message tab (`/messages`) are not filtered by the currently active language. I think it would be helpful to add this filtering because this view allows quick and simple message feedback (e.g. emojis & reporting). When many messages are shown in foreign languages which the current user does not understand we lose potential message-review capacity...

The queries for the messages should send the lang-tag of the currently active language as query param to the backend. When the user switches the language (via language selector drop down) the message result lists should be refreshed.

**Comment by jebbbbb** [comment]
Hey, I'm a research engineer currently working on a mix of ml research and software engineering with a focus on language modelling. I'd like to get involved and take this issue if it's still free?

**Comment by fozziethebeat** [comment]
Hi @jebbbbb have you worked with NextAuth localization?  I'm happy to let you take a look at this if you want although I setup this part and I _think_ I know the quick fix to it.

**Comment by jebbbbb** [comment]
Hey @fozziethebeat, no I haven't used it before. In that case it probably makes more sense for you to take it. I also noticed #990 is still open and think I've already got an idea how I would solve it, could you assign me to that one?

**Comment by jackapbutler** [comment]
Hey @fozziethebeat, I've changed GitHub accounts to my personal username. No, I haven't worked with NextAuth localization before. In that case it probably makes more sense for you to take it. I'd be happy to take up another ticket like #990 if you could assign me to it?

**Comment by fozziethebeat** [comment]
I'll get to this shortly

**Comment by fozziethebeat** [comment]
@andreaskoepf to make this work, method `query_messages_ordered_by_created_date` in `prompt_repository.py` has to be updated.  The lang filtering has to happen before ordering (I just tested locally and it currently fails when you fetch recent messages with the lang field set).

I tried to make this in my local PR but for some reason the `check python ast` pre-commit is flagging this file because it uses match (which I guess isn't supported in CPython 3.9.13 that's being run).

Can you fix the backend side? I have a web side fix that'll request it in the user's language for the short term.
